### DISCUSSION TITLE
execute sequel model hooks after the account was created

### DISCUSSION THREAD

**Discussion by opya** [jeremyevans/rodauth#345] [General]
Hi guys,

I was wondering how to execute defined model hooks when creating the account. Currently, I have this hook

```ruby
class Account < Sequel::Model
  def before_create
    self.settings = DEFAULT_SETTINGS
    super
  end
end
```

I understand that `rodauth` doesn't use the model but I'm curious how to execute the above hooks after the account was created.
Can change the above hook to `before_save` and check if the settings are already set, but I'm curious where is a good place to do this after the account was created. Thinking to use the `after_create_account` rodauth hook and save it again, but I'm curious if there is a better way, how you do such things ? :)

**Comment by jeremyevans** [comment]
You could have the Rodauth hooks create a model and call the appropriate method on it.  However, this may work in your Rodauth config and be simpler:

```ruby
  new_account do
    super.merge!(settings: Account::DEFAULT_SETTINGS)
  end
```

**Reply by opya** [reply]
Thank you Jeremy, it works perfectly.
